bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/loader/AbstractScriptFileWatcher.java

```diff
@@ -144,6 +146,9 @@ public void activate() {
         } else if (!Files.isDirectory(watchPath)) {
             logger.warn("Trying to watch directory {}, however it is a file", watchPath);
         }
+
+        manager.addFactoryChangeListener(this);
+        readyService.registerTracker(this, new ReadyMarkerFilter().withType(StartLevelService.STARTLEVEL_MARKER_TYPE));
```

Comment:
Why did you move these?

Reply:
Remember that readyService.registerTracker will catch you up on all the startlevel markers that had already been received/reached? Well, that would trigger initialImport right there if I left it in the constructor.

Reply:
Indeed. At first I only wanted to avoid the NPE and didn't want to change the existing code much. But you're correct, there's no need to do initialImport twice. I've removed it.